### ISSUE TITLE
The Yeaoman @microsoft/generator-sharepoint@1.4.0 comes with no scaffold support for tests files

### ISSUE THREAD

**Issue by VelinGeorgiev** [SharePoint/sp-dev-docs#1335]
#### Category
- [X] Question
- [ ] Typo
- [ ] Bug
- [ ] Additional article idea

The Yeaoman @microsoft/generator-sharepoint@1.4.0 comes with no scaffold support for tests files for React and JavaScript client side solution templates. That feature was present in older versions and was useful to 3 teams within my organization.

What might be the reason for removing that from a newly created web part scaffolded by the Yeaoman generator?

Maybe option for including the test folder and test file in the solution upon creation could be beneficial. Something similar to Visual Studio that has check box allowing to include Tests project in your C# solution.

**Comment by VesaJuvonen** [comment]
Thx @VelinGeorgiev  for reporting, let me follow up on this internally. We should definitely not remove anything without proper documentation in release notes.

**Comment by VelinGeorgiev** [comment]
@VesaJuvonen , @patmill do you know what would be the future of the SharePoint testing framework and packages so we can eventually plan migration if that would be deprecated.

**Comment by waldekmastykarz** [comment]
Ideally, wouldn't it be the best if the SharePoint Framework allowed us to use any testing framework we want? There are quite a few testing frameworks available out there and everyone has their own preference. As long as things just work and developers can use the testing framework of their choice, I'd say it's not necessary for Yeoman to scaffold tests by default.

**Comment by VelinGeorgiev** [comment]
I agree with that. It is just we had one setup provided OOTB and we decided to use it in the past. If this is going to be deprecated, it would require technological time for us to migrate and that has to be communicated.

**Comment by tracy-zaact** [comment]
@waldekmastykarz - Will you create a user voice entry for your recommendation?  Myself and my colleagues will up vote it.

**Comment by baywet** [comment]
stumbled on this issue recently trying to debug unit tests with 1.4.1 and see what's missing. Agree with the fact that spfx should be "test framework agnostic" and let the community provide samples for different options. My original post from almost a year ago https://github.com/SharePoint/sp-dev-docs/issues/736

**Comment by VelinGeorgiev** [comment]
I also agree with the fact that SPFx should be test framework agnostic and infarct we already have PnP sample and examples how the Jest testing framework can be setup. 

I decided to use the Jest instead of the OOTB provided by the #SPFx. That worked for me and I use it in all my projects now. 

There is a PnP sample http://bit.ly/2Hiy4qb , but you can find guidance at http://bit.ly/2T08ho1  or http://bit.ly/2szHzXO  or http://bit.ly/2QUZ4LR
